parquet-hadoop/src/main/java/org/apache/parquet/ParquetReadOptions.java

```diff
@@ -44,6 +46,9 @@ public class ParquetReadOptions {
   private static final int ALLOCATION_SIZE_DEFAULT = 8388608; // 8MB
   private static final boolean PAGE_VERIFY_CHECKSUM_ENABLED_DEFAULT = false;
   private static final boolean BLOOM_FILTER_ENABLED_DEFAULT = true;
+  // Default to true if JDK 17 or newer.
+  private static final boolean USE_OFF_HEAP_DECRYPT_BUFFER_DEFAULT =
+    SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_17);
```

Comment:
Will direct buffers have better performance than byte arrays or indirect byte buffers? Or maybe the default can be off; with "on" to be set explicitly by users?

Reply:
Yep, lets do it. Given the current state of byte buffer decryption in Java (not all problems are resolved in JVM17), it'd be safer to keep this off, and let the users set to on explicitly.